ed25519/src/seed.cpp

```diff
@@ -3,11 +3,18 @@
 
 #include <boost/system/error_code.hpp>
 #include <boost/system/system_error.hpp>
+#include "libtorrent/config.hpp"
 #include "libtorrent/ed25519.hpp"
 
 #ifndef ED25519_NO_SEED
 
-#ifdef _WIN32
+#ifdef TORRENT_WINRT
```

Comment:
Instead of using ifdefs based on operating systems, I would prefer having switches (defines) that enable/disable specific APIs. The config.hpp would then pick the APIs supported for a specific system. This is more flexible if there are some variation (think mingw, win32 and winrt. they share a lot of APIs, but there are also a lot only supported by one of them). In short, I think there should be a define along the lines of: TORRENT_USE_CRYPTOGRAPHIC_BUFFER which enables the use of this API.